### ISSUE TITLE
[bug]  Not clear conan work with defaul_options

### ISSUE THREAD

**Issue by AlekseyKovy** [conan-io/conan#7710]
In some cases, the conan ignore default_options in conanfile.py

### Environment Details 
  * Operating System+version: Any
  * Compiler+version: Any
  * Conan version: 1.28
  * Python version:  3.8

### Steps to reproduce 
Local cache contains 1 prebuilt package dep1 with default_options(dep1:option1=False).

```python3
from conans import ConanFile

class DepConan(ConanFile):
    name = "dep1"
    version = "0.0.1"
    exports_sources = "src/*"
    settings = "os", "build_type"
    options = {"option1":[True, False]}
    default_options = {"option1": False}
```

We have an application recipe that depends on dep1:option1=True.
```python3
from conans import ConanFile


class AppConan(ConanFile):
    name = "app"
    version = "0.0.1"
    exports_sources = "src/*"
    settings = "os", "build_type"
    options = {"option1":[True, False] }
    requires = ["dep1/0.0.1@integration_tests/case12"]
    default_options = { "dep1:option1": True  }
```
When trying to build an application with default options, conan uses dep1:option1=False instead of dep1:option1=True.

Lockfile application, which was built with default options:
``` json
    "nodes": {
      "0": {
        "ref": "app/0.0.1@integration_tests/case12",
        "options": "option1=False\ndep1:option1=False",
        "requires": [
          "1"          
        ],
        "context": "host",
        "modified": true,
        "build_requires": null
      },
      "1": {
        "ref": "dep1/0.0.1@integration_tests/case12#502d5b56a23e26bbea62e1e4a010aa84",
        "options": "option1=False",
        "context": "host",
        "modified": true,
        "requires": null,
        "build_requires": null
      }
    }
  },
```

What is the logic of conan when it works with default_options?

**Comment by memsharded** [comment]
Hi @Alex0Lihach 

Could you please detail what are the steps, the full conan commands that you are using (including details about the profiles if necessary), and the output of those commands?
I am trying to reproduce with the following test, but it works fine:

```python
    def default_options_test(self):
        # https://github.com/conan-io/conan/issues/7710
        client = TestClient()
        conanfile = textwrap.dedent("""
            from conans import ConanFile

            class DepConan(ConanFile):
                options = {"option1":[True, False]}
                default_options = {"option1": False}
            """)
        client.save({"conanfile.py": conanfile})
        client.run("create . dep/0.1@ -o dep:option1=True")
        
        conanfile = textwrap.dedent("""
            from conans import ConanFile

            class AppConan(ConanFile):
                options = {"option1":[True, False] }
                requires = "dep/0.1"
                default_options = { "dep:option1": True, "option1": True }
            """)
        client.save({"conanfile.py": conanfile})
        client.run("create . app/0.1@")
        client.run("lock create --reference=app/0.1 --lockfile-out=conan.lock")
        lock = client.load("conan.lock")
        self.assertIn("dep:option1=True", lock)
```

**Comment by AlekseyKovy** [comment]
```sh
client.run("create . dep/0.1@ -o dep:option1=True")
```
You build dep with -o dep:option1=True, but I build dep without -o flag(just "conan create ." in dep folder). After building, my local cache has 1 dep package with option1=False. And now when I try to build app("conan create ." in app folder), conan builds it with dep:option1=False, in fact, ignoring default_option

**Comment by memsharded** [comment]
Hi @Alex0Lihach 

If I remove the ``-o dep:option1=True``, it is created with default value False, that is right.
But then, when I later do ``create . app/0.1@``, it fails with a *"missing binary error"*. Conan does not build it by default. If I pass ``--build=missing`` to ``conan create . app/0.1 --build=missing``, the test is green again.

Could you please detail the complete sequence of commands that you are using? Many thanks!

**Comment by AbrilRBS** [comment]
Closing as we didn't hear back from you in quite a lot of time, of anyone thinks they are in a similar situation, please feel free to create a new issue, thanks!
